src/lib.rs

```diff
@@ -118,39 +118,42 @@ where
     _marker: PhantomData<R>,
 }
 
-impl<R> Default for MenuStyle<BinaryColor, LineIndicator, Programmed, StaticPosition, R> {
+impl<R> Default
+    for MenuStyle<LineIndicator<BinaryColor>, Programmed, StaticPosition, R, BinaryColor>
+{
     fn default() -> Self {
-        Self::new(BinaryColor::On)
+        Self::new(BinaryColor::On, BinaryColor::On)
     }
 }
 
-impl<C, R> MenuStyle<C, LineIndicator, Programmed, StaticPosition, R>
+impl<IC, MC, R> MenuStyle<LineIndicator<IC>, Programmed, StaticPosition, R, MC>
 where
-    C: PixelColor,
+    IC: PixelColor,
+    MC: PixelColor,
 {
-    pub const fn new(color: C) -> Self {
+    pub const fn new(menu_color: MC, indicator_color: IC) -> Self {
```

Comment:
I'm not sure why we need two color types. I don't think different types would compile for IC and MC?

So what I think we should do is define a Theme of sorts, instead of inserting every color one-by-one here.

```rust
trait Theme {
    type Color: PixelColor;

    fn text_color(&self) -> Self::Color;
    fn inverted_text_color(&self) -> Self::Color; // Inverted referring to the text under the progress display part of the indicator
    fn indicator_color() -> Self::Color;
... we add more as needed
}
```

This would be easy to implement for BinaryColor by default and maybe we could provide named Themes for other color types.

Reply:
This is admittedly an artifact of an original pass attempting to be as flexible with color spaces as possible on a per-style component basis. These two colors can be different, with the only restriction being that each can be converted into the color space used by the display. 

A lot of this work was walked back as it felt unnecessarily complex and was a larger change than I think was actually necessary or desired, especially for this first pass. I'd be happy to switch these both to use the same color space

Reply:
I have to admit I don't see a point to two color spaces for two different elements on the screen. It will all boil down to a single color in the end and conversion steps in-between would just make things more complicated than necessary. I hardly see reason to allow setting text color to `BinaryColor::On` with an RGB888 indicator color, even though some conversion could probably make something of these two. From a user's perspective this is very confusing.

Reply:
very fair point. Where my mind was originally at with this was looking to allow people to write crates that act as extensions of `embedded-menu` through implementation of the `MenuItem` trait. However, I feel this comes with multiple undesirable side effects such as complicating the code within `embedded-menu` as well as allowing extension crate authors to be lazy with their menu item's color space. 

In short - I think this PR can be updated to resolve to a single mono-color space